classes/setup_item_processor.php

```diff
@@ -81,78 +85,192 @@ public function process(setup_item $item) {
         }
 
         foreach ($courses as $course) {
-            $page = new \moodle_page();
-            $page->set_course($course);
-            $page->set_pagelayout('course');
-            $page->set_pagetype('course-view-' . $course->format);
-
-            if (!$page->blocks instanceof block_manager) {
-                // TODO: disable whole plugin if block manager is not overridden.
-                throw new \coding_exception('Terminate processing. Block manager class is not configured in config.php');
-            }
 
-            try {
-                $page->blocks->add_region($item->get_region(), false);
-                $page->blocks->load_blocks(true);
-                $page->blocks->create_all_block_instances();
+            $pagetypesegments = explode('-', $item->get_pagetypepattern());
+
+            // Check if block is to be added to a course module.
```

Comment:
Would be nice to document this new feature in README as well as possibly update help text for "Page type pattern" and give few examples of patterns that can be used .

Reply:
This has been added in the following commit: https://github.com/catalyst/moodle_tool_blocksmanager/pull/45/commits/e0763ae750785966fe068cab000b094735d77a03

This included listing the new feature and examples of the page type patterns that can be used.